tests/test_cyclic_topology.py

```diff
@@ -0,0 +1,55 @@
+import os
+import stk
+
+test_dir = 'cyclic_topology_tests_output'
+if not os.path.exists(test_dir):
+    os.mkdir(test_dir)
+
+
+def test_assembly(amine2, aldehyde2, boronic_acid2, diol2):
+    repeat_units = 3
+    monomer_joins = 2*repeat_units
+
+    c1 = stk.Macrocycle([amine2, aldehyde2],
+                     stk.Cyclic('AB', [0, 0], repeat_units))
+
+    c2 = stk.Macrocycle([boronic_acid2, diol2],
+                     stk.Cyclic('AB', [0, 0], repeat_units))
+
+    path = os.path.join(test_dir, 'c1.mol')
+    c1.write(path)
+    c2.write(path.replace('1', '2'))
+
+    assert c1.bonds_made == monomer_joins
+    assert c2.bonds_made == monomer_joins*2
+
+    monomer_atom_count = (amine2.mol.GetNumAtoms() +
+                          aldehyde2.mol.GetNumAtoms())
+
+    # 3 atoms are lost at each join in c1 and c2 due to condensation.
+    assert (c1.mol.GetNumAtoms() ==
+            monomer_atom_count*repeat_units - 3*monomer_joins)
+
+    # p3 loses 2 atoms becaus OH groups are replaced with H.
```

Comment:
If you replace OH with H do you not lose 1 atom?

Reply:
There are 2 OHs

Reply:
It would make sense to add it to the comment to ensure I am not confused in the future.